build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildDatabase.ts

```diff
@@ -0,0 +1,346 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import path from "node:path";
+
+import { type Package, TscUtils } from "@fluidframework/build-tools";
+import type { TsConfigJson } from "type-fest";
+
+import { ApiLevel } from "../apiLevel";
+import { ApiTag } from "../apiTag";
+import { queryOutputMapsFromPackageExports } from "../packageExports";
+
+type PackageName = string;
+type Script = string;
+/**
+ * `${package.name}#${script}`
+ */
+export type PackageNameAndScript = string;
+type AbsoluteFilePath = string;
+type ModuleType = "CommonJS" | "ESM";
+export interface BuildScript {
```

Comment:
Docs please (plus members). In particular, what does it mean for `moduleType` to be undefined?